src/runtime_init_memory.js

```diff
@@ -91,3 +91,18 @@ HEAP32[DYNAMICTOP_PTR>>2] = DYNAMIC_BASE;
 #if USE_PTHREADS
 }
 #endif
+
+#if MIN_CHROME_VERSION < 45 || MIN_EDGE_VERSION < 14 || MIN_FIREFOX_VERSION < 38 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED
+// Older browsers may lack TypedArray.slice()
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice#Browser_compatibility
+[Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array,
```

Comment:
Note that all of those actually have a super-class `TypedArray` that is not directly exposed. That is, instead of adding a method to each and every class, you can get a `TypedArray` reference via

```js
var typedArrayProto = Int8Array.prototype.__proto__;
```

and then polyfill `typedArrayProto.slice` once.

Reply:
Interesting, thanks, I didn't know that.

Do you know if that's well-supported in all legacy browsers?

Reply:
I don't, unfortunately, and can't test it.

Reply:
_Still relevant. In fact, we need [these](https://github.com/libass/JavascriptSubtitlesOctopus/blob/6e74ea2ced17e4364026e9f6dd9f3027ecdc83c5/src/pre-worker.js#L1-L50) polyfills._

Tested this polyfill:
```js
var typedArrayProto = Object.getPrototypeOf(Int8Array.prototype);
if (typedArrayProto && !typedArrayProto.slice) {
    typedArrayProto.slice = function (begin, end) {
        return new this.constructor(this.subarray(begin, end));
    }
}
```
**webOS 1.2 (WebKit)**
Polyfill works correctly.
```
Object.getPrototypeOf(Int8Array.prototype):     [object ArrayBufferViewPrototype]
Int8Array.prototype:                            [object Int8ArrayPrototype]
Object.getPrototypeOf(Int8Array):               [object Object]
Int8Array:                                      [object Int8ArrayConstructor]

Object.getPrototypeOf(Int32Array.prototype):    [object ArrayBufferViewPrototype]
Int32Array.prototype:                           [object Int32ArrayPrototype]
Object.getPrototypeOf(Int32Array):              [object Object]
Int32Array:                                     [object Int32ArrayConstructor]

Object.getPrototypeOf(Int8Array.prototype) === Object.getPrototypeOf(Int32Array.prototype): true
Object.getPrototypeOf(Int8Array.prototype) === Object.getPrototypeOf({}): false
```

**webOS 3 (Chrome 38)**
Polyfill works, but `{}` gets `slice`. So it is probably better to leave as it is.
```
Object.getPrototypeOf(Int8Array.prototype):     [object Object]
Int8Array.prototype:                            [object Object]
Object.getPrototypeOf(Int8Array):               function Empty() {}
Int8Array:                                      function Int8Array() { [nativecode] }

Object.getPrototypeOf(Int32Array.prototype):    [object Object]
Int32Array.prototype:                           [object Object]
Object.getPrototypeOf(Int32Array):              function Empty() {}
Int32Array:                                     function Int32Array() { [nativecode] }

Object.getPrototypeOf(Int8Array.prototype) === Object.getPrototypeOf(Int32Array.prototype): true
Object.getPrototypeOf(Int8Array.prototype) === Object.getPrototypeOf({}): true
```

Reply:
@dmitrylyzo thanks for the info. Would you like to open a PR that takes this code and updates it to use the right polyfills?

Reply:
I tested your polyfill - works fine. As I said, I think it is better (for legacy browsers) than the super-class polyfill, because in webOS 3 super-class polyfill adds `slice` to every object due to a common prototype (but it could just be a glitch of the emulator).

As for the other polyfills, they are for the libraries used in that project. Wouldn't their inclusion bloat the output file?

Reply:
Can I ask what targets do you have that run webOS 3 (chrome 38).   That seems like a rather old version of the browser to target.

Reply:
Mostly old TVs (https://webostv.developer.lge.com/discover/specifications/web-engine/). Jellyfin is targeting Chrome 27 as a rough approximation of webOS 1.2 and 2.0 (both WebKit).